internal/backend/secrets/svc.go

```diff
@@ -3,29 +3,54 @@ package secrets
 import (
 	"context"
 	"fmt"
+	"strings"
+	"time"
 
 	"github.com/lithammer/shortuuid"
 	"go.uber.org/zap"
 	"golang.org/x/exp/maps"
 
+	"go.autokitteh.dev/autokitteh/internal/backend/configset"
 	"go.autokitteh.dev/autokitteh/internal/xdg"
 	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
 )
 
-type secrets struct {
-	impl   Secrets
-	logger *zap.Logger
+type Config struct {
+	ManagerURL      string `koanf:"manager_url"`
+	TimeoutDuration string `koanf:"timeout_duration"`
 }
 
-func New(l *zap.Logger) (sdkservices.Secrets, error) {
-	// TODO(ENG-145): Allow the user to select the implementation via CLI.
-	impl, err := NewFileSecrets(l, xdg.DataHomeDir())
+var Configs = configset.Set[Config]{
+	Default: &Config{},
+}
+
+var defaultTimeout = 1 * time.Minute
+
+func New(l *zap.Logger, cfg *Config) (sdkservices.Secrets, error) {
+	var impl Secrets
+	var err error
+
```

Comment:
ParseURL

Reply:
I had to add a type field, which made parsing the URL unnecessary anyway.

Reply:
it could have just been the scheme, but oh well